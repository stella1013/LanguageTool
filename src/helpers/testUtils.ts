//import { ConnectedRouter } from 'connected-react-router';
//import React from 'react';
//import { Provider } from 'react-redux';
////import { mergeDeepRight } from 'ramda';
//import rootReducer from '../redux/reducers';
////import  createStoreWithMiddleWare  from '../redux/store';
//import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
//import { AppState } from '../redux/store';
//import ReduxThunk  from 'redux-thunk';
//import { createStore, applyMiddleware } from 'redux';
//import rootReducer from '../redux/reducers';
//import thunk from 'redux-thunk';

/*
	* Returns Shallow Wrapper containing node(s) with the given data-test value
	* @param {Shallow Wrapper} - Enzyme shallow wrapper to search within
	* @param {string} val - Value of data-test attribute for search.
	* @returns {Shallow Wrapper}
	*/ 
	export const findByTestAttr = (wrapper:any, val:string) => {
		return wrapper.find(`[data-test="${val}"]`);
	}

	export const storeFactory = (initialState:any) =>{
		const createMockStore = configureMockStore();
		return createMockStore(initialState);
	}
	/*
	export const makeMountRender = (Component:any, defaultProps = {}) => {
		return (customProps = {}) => {
		  const props = {
			...defaultProps,
			...customProps
		  };
		  return mount(<Component {...props} />);
		};
	  };

	  export const makeStore = (customState = {}) => {
		const root = rootReducer({}, { type: '@@INIT' });
		const state = mergeDeepRight(root, customState);
		return createStoreWithMiddleWare(rootReducer, state);
	  };
	  export const snapshotify = (reactWrapper:any) => {
		return reactWrapper.html();
	  };

	  export const reduxify = (Component, props = {}, state = {}) => {
		return function reduxWrap() {
		  return (
			<Provider store={makeStore(state)}>
			  <ConnectedRouter history={history}>
				<Component {...props} />
			  </ConnectedRouter>
			</Provider>
		  );
		}
	  };

*/