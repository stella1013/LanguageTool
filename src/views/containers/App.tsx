import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../redux/store';
import { Container, Navbar } from 'react-bootstrap';
import { dataOperations } from '../../redux/Data';
import DashboardContainer from './DashboardContainer';
import EditorContainer from './EditorContainer';
import PreviewContainer from './PreviewContainer';
import SuccessContainer from './SuccessContainer';
import { modalOperations } from '../../redux/Modal';
import { appTitleOperations, appTitleSelectors } from '../../redux/AppTitle';
import { ScreenTypes } from '../../redux/ModeController/constants';
import AuthContainer from './Auth/AuthContainer';
import UserProfileContainer from './UserProfileContainer';
import { modeOperations } from '../../redux/ModeController';

/*
TODO:Add the following reducers
	isReady: false,?
	currentPage: 0,
	totalPages: 999999,
	pages: [],
	selected/inputed data object?
*/
class App extends React.Component<ReduxType> {
	currentMode: any;
	constructor(props:any){
		super(props);
		this.props.getData();
		this.props.setAppTitle('Asset Ingesting Application');
	}
	componentDidMount = () => {
		
	};
	showUserProfile = (e:any) =>{
		this.props.changeAppMode(ScreenTypes.USERPROFILE_MODE);
	}
	getCurrentMode = (mode: ScreenTypes) => {
		if(this.props.isAuth === true){
			if (mode === ScreenTypes.EDIT_MODE) {
				this.currentMode = <EditorContainer data-test="dashboard-mode"/>;
			} else if (mode === ScreenTypes.PREVIEW_MODE) {
				this.currentMode = <PreviewContainer />;
			} else if (mode === ScreenTypes.SUCCESS_MODE) {
				this.currentMode = <SuccessContainer />;
			} else if (mode === ScreenTypes.USERPROFILE_MODE) {
				this.currentMode = <UserProfileContainer />;
			} else {
				this.currentMode = <DashboardContainer />;
			}
		} else {
			this.currentMode = <AuthContainer/>;
		}
		return this.currentMode;
	};

	render() {
		return (
			<div>
			<Navbar bg="dark" variant="dark">
				<Navbar.Brand href="#home">
				{this.props.isAuth ? ' User Name' : this.props.appTitle}
				</Navbar.Brand>
				<Navbar.Collapse className="justify-content-end">
					<Navbar.Text>
					{this.props.isAuth ? ( <div>You are logged In:&nbsp; <button onClick={(e:React.MouseEvent)=>this.showUserProfile(e)}>Settings</button></div> )
					: <a href="#login">Log In</a> }
					</Navbar.Text>
				</Navbar.Collapse>
				
			</Navbar>
			<Container data-test="component-app" >
				{this.getCurrentMode(this.props.modeController)}
			</Container>
			</div>
		);
	}
}
const mapStateToProps = (state: AppState) => {
	return {
		data: state.data.pages,
		modeController: state.modeController.currentMode,
		isAuth: state.appAuth.isAuth,
		appTitle: appTitleSelectors.getAppTitleSelectorMemo(state)
	};
};
const mapDispatchToProps = (dispatch: any) => {
	return {
		getData: () => dispatch(dataOperations.fetchData()),
		showModal: (showModal: boolean) => dispatch(modalOperations.showModal(showModal)),
		setAppTitle: (title: string) => dispatch(appTitleOperations.setAppTitle(title)),
		changeAppMode: (newMode:ScreenTypes) => dispatch(modeOperations.changeMode(newMode))
	};
};
type ReduxType = ReturnType<typeof mapStateToProps> &
	ReturnType<typeof mapDispatchToProps>;
export default connect( mapStateToProps, mapDispatchToProps)(App);
