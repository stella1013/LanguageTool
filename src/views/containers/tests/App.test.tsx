
import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import { findByTestAttr } from '../../../helpers/testUtils';
import App from '../App';
import DashBoardContainer from '../DashboardContainer';

describe('App', () => {
	/*
	 * Factory function to create a Shallow Wrapper for the App Component
	 * @function setup
	 * @param {object} props - component props
	 * @param {object} state - initial state
	 * @returns {Shallow Wrapper}
	 */

	const createMockStore = configureMockStore([thunk]);
	const initialState = {};
	//const store = createMockStore(initialState);
	let wrapper: any;
	describe('rendering', () => {
		//const mockFetchData = jest.fn();
		//const props = {  }
		beforeEach(() => {
			const store = createMockStore(initialState);
			wrapper = shallow(<Provider store={store}><App /></Provider>);
				
		});
		

		it('renders without crashing', () => {
			const appComponent = findByTestAttr(wrapper, 'component-app');
			expect(appComponent).toBeTruthy();
		});

		it('render matches snapshot', () => {
			expect(wrapper).toMatchSnapshot();
		});
		
		describe('rendering different screens', () => {
			
			it('initial load renders DashBoard Mode', () => {
				expect(wrapper.find(<DashBoardContainer />)).toBeTruthy();
			});
		});

	});

	describe('when user updates `state`', () => { 
		
	});
});

