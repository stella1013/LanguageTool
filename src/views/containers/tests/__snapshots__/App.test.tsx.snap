// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering render matches snapshot 1`] = `ShallowWrapper {}`;
